tests/run_bar_tests.sh

```diff
@@ -0,0 +1,199 @@
+		sed -r -i -e '1d' "$log_file"
+	fi
+	local line="$(head -n1 "$log_file")"
+	local linenum="$(cat "$log_file" | wc -l)"
+	if test $linenum -lt 5 ; then
+		fail "log:lt" F "Script was run not enough times"
+		return 1
+	elif test $linenum -gt 15 ; then
+		fail "log:gt" E "Script was run too many times"
+		return 1
+	fi
+	local expline="%{l}%{F#ffd0d0d0}%{B#ff303030} $text-left %{F-B--u}%{F#ff303030} %{F-B--u}%{r}%{F#ff303030} %{F-B--u}%{F#ffd0d0d0}%{B#ff303030} $text-right %{F-B--u}"
+	if test "x$expline" != "x$line" ; then
+		echo "Line:     '$line'"
+		echo "Expected: '$expline'"
+		fail "log:line" F "Unexpected line"
+		return 1
+	fi
+	local ret=0
+	while test $linenum -gt 0 ; do
+		echo "$line" >> "$TEST_ROOT/ok"
+		linenum=$(( linenum - 1 ))
+	done
+	if ! diff "$TEST_ROOT/ok" "$log_file" ; then
+		fail "log:diff" F "Unexpected output"
+		ret=1
+	fi
+	rm "$TEST_ROOT/ok"
+	return $ret
+}
+
+killscript() {
+	kill -KILL $1 || true
+}
+
+if ! test -e "$DEPRECATED_SCRIPT" ; then
+	# TODO: uncomment when skip is available
+	# skip "deprecated" "Missing deprecated bar bindings script"
+	:
+else
+	enter_suite "deprecated"
+	run python "$DEPRECATED_SCRIPT" $args &> "$TEST_ROOT/deprecated.log" &
+	SPID=$!
+	sleep 5
+	killscript $SPID
+	if ! check_log "$TEST_ROOT/deprecated.log" "default" warns ; then
+		display_log "$TEST_ROOT/deprecated.log"
+		fail "log" F "Checking log failed"
+	fi
+	rm "$TEST_ROOT/deprecated.log"
+	exit_suite --continue
+fi
+
+LEMONBAR_SCRIPT="$ROOT/powerline/bindings/lemonbar/powerline-lemonbar.py"
+
+if ! test -e "$LEMONBAR_SCRIPT" ; then
+	# TODO: uncomment when skip is available
+	# skip "lemonbar" "Missing lemonbar bindings script"
+	:
+else
+	enter_suite "lemonbar"
+	for args in "" "-i0.5" "--interval=0.5" "-- test args" "--height=10" "-c bar-aint-recursive"; do
+		rm -rf "$TEST_ROOT/results"
+		run python "$LEMONBAR_SCRIPT" $args &> "$TEST_ROOT/lemonbar.log" &
+		SPID=$!
+		sleep 5
+		killscript $SPID
+		sleep 0.5
+		enter_suite "args($args)"
+		fnum=0
+		for file in "$TEST_ROOT/results"/*.log ; do
+			if ! test -e "$file" ; then
+				fail "log" E "Log file is missing"
+				break
+			fi
+			fnum=$(( fnum + 1 ))
+			args_file="${file%.log}.args"
+			if ! test -e "$args_file" ; then
+				fail "args" E "$args_file is missing"
+			else
+				cat "$args_file" >> "$TEST_ROOT/args.log"
+			fi
+			text="dvi"
+			if cat "$args_file" | grep -q +1 ; then
+				text="default"
+			fi
+			if ! check_log "$file" "$text" ; then
+				display_log "$file"
+				fail "log" F "Checking log failed"
+			fi
+			rm "$file"
+		done
+		if test "$fnum" -ne 2 ; then
+			fail "fnum" F "Expected two output files"
+		fi
+		if test "x${args#--height}" != "x$args" ; then
+			height="${args#--height}"
+			height="${height# }"
+			height="${height#=}"
+			height="${height%% *}"
+		fi
+		command="lemonbar"
+		if test "x${args#-c}" != "x$args" ; then
+			command="${args#-c}"
+			command="${command# }"
+			command="${command#=}"
+			command="${command%% *}"
+		fi
```

Comment:
No, it is OK to try only `--bar-command` (*not* `-C`, long versions of arguments are always preferred and if you want to test only one variant test only long). But rename `-c` to `-C` in case somebody will actually add overrides (`-c` stands for `--config-override` in `powerline`/`powerline-render`).